src/lightning_lite/strategies/fsdp.py

```diff
@@ -291,6 +309,22 @@ def _set_world_ranks(self) -> None:
         rank_zero_only.rank = self.cluster_environment.global_rank()
 
 
+def _setup_activation_checkpointing(module: "FullyShardedDataParallel", layers: List[Type[Module]]) -> None:
+    from torch.distributed.algorithms._checkpoint.checkpoint_wrapper import (
+        apply_activation_checkpointing,
+        checkpoint_wrapper,
+        CheckpointImpl,
+    )
+
+    check_fn = lambda submodule: isinstance(submodule, tuple(layers))
+    wrapper = functools.partial(
+        checkpoint_wrapper,
+        offload_to_cpu=False,
```

Comment:
What would be the disadvantage of leaving CPU offloading as an option?

Reply:
I just explicitly wrote it to signal that this is the default setting in FSDP. This is specific to version 1.13. The argument was already removed on master (1.14 nightly) and so I wouldn't expose it now unless their API becomes stable.